[Doc] Ray Debugging Doc Part 1 (OOM) (#34309)
ray-project/ray: doc/source/ray-core/scheduling/memory-management.rst

Application memory: this is memory used by your application
  - **Worker heap**: memory used by your application (e.g., in Python code or TensorFlow), best measured as the *resident set size (RSS)* of your application minus its *shared memory usage (SHR)* in commands such as ``top``. The reason you need to subtract *SHR* is that object store shared memory is reported by the OS as shared with each worker. Not subtracting *SHR* will result in double counting memory usage.

>   - **Object store memory**: memory used when your application creates objects in the object store via ``ray.put`` and when returning values from remote functions. Objects are reference counted and evicted when they fall out of scope. There is an object store server running on each node. By default, 30% of available memory when starting a Ray instance is reserved to ``/dev/shm`` (shared memory) in Linux. In MacOS, it uses ``/tmp``, which can impact the performance compared to Linux (because ``/dev/shm`` uses the shared memory, but ``/tmp`` uses the disk). In Ray 1.3+, objects will be :ref:`spilled to disk <object-spilling>` if the object store fills up.

Reviewer: ```suggestion
  - **Object store memory**: memory used when your application creates objects in the object store via ``ray.put`` and when it returns values from remote functions. Objects are reference counted and evicted when they fall out of scope. An object store server runs on each node. By default, when starting an instance, Ray reserves 30% of available memory on ``/dev/shm`` (shared memory) for Linux. For MacOS, Ray uses ``/tmp`` (disk), which can impact the performance compared to Linux. In Ray 1.3+, objects are :ref:`spilled to disk <object-spilling>` if the object store fills up.
```

Reviewer: See if you like the rewrite. I removed the phrase in the parenthesis, assuming that it didn't add information if you added `(disk)` to the first mention of `/tmp`.
